src/org/zaproxy/zap/extension/ascanrulesAlpha/HiddenFileScanner.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.*;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractHostPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+
+
+/**
+ * An example active scan rule, for more details see
+ * http://zaproxy.blogspot.co.uk/2014/04/hacking-zap-4-active-scan-rules.html
+ * @author psiinon
+ */
+public class HiddenFileScanner extends AbstractHostPlugin {
+
+    private static final String MESSAGE_PREFIX = "ascanalpha.hiddenfilescanner.";
+    private static final int PLUGIN_ID = 40039;
```

Comment:
The ID should be added to:
https://github.com/zaproxy/zaproxy/blob/develop/docs/scanners.md

Reply:
Should I create a separate PR for that in zaproxy?

Reply:
Yes, please

Reply:
Created a PR :)